models/migrations/v211.go

```diff
@@ -0,0 +1,248 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"code.gitea.io/gitea/models/auth"
+	"code.gitea.io/gitea/models/perm"
+	"code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/models/user"
+	api "code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+)
+
+func convertFromNullToDefault(x *xorm.Engine) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	type Label struct {
+		ID              int64 `xorm:"pk autoincr"`
+		RepoID          int64 `xorm:"INDEX"`
+		OrgID           int64 `xorm:"INDEX"`
+		Name            string
+		Description     string
+		Color           string             `xorm:"VARCHAR(7)"`
+		NumIssues       int                `xorm:"NOT NULL DEFAULT 0"`
+		NumClosedIssues int                `xorm:"NOT NULL DEFAULT 0"`
+		CreatedUnix     timeutil.TimeStamp `xorm:"INDEX created"`
+		UpdatedUnix     timeutil.TimeStamp `xorm:"INDEX updated"`
+	}
+	recreateTable(sess, &Label{})
```

Comment:
is there a way to only focus on adding the defaults to the relevant columns?

Reply:
That was the easiest way I found to support all SQL dialects. @lunny are there (short) alternatives with xorm?

Reply:
We could use `modifyColumn` directly which could change the default value

Reply:
`// modifyColumn will modify column's type or other propertity. SQLITE is not supported`
SQlite does not support `ALTER COLUMN`.

Reply:
Yes, for SQLITE, maybe `recreate` is the only way.

Reply:
we could check what dbms is used and use recreate only for sqlite?

Reply:
Modify column will be much faster migration for large instances, for SQLite it would be fine with recreate tables as most commonly they should be small instances

Reply:
The migration runs a single time. Is speed or readability the goal?

Reply:
What's "too long"? Did you measure if it is really "too long"?

Reply:
That's the reason why I asked if it got measured if it really takes hours. Anyway, MySQL and Postgresql alter the columns directly now.

Reply:
JFI: I don't know how big codeberg database is but for large databases it could be not only speed problem but also space